site-src/geps/gep-713.md

```diff
@@ -0,0 +1,409 @@
+* May be difficult to understand which policies apply to a request
+
+
+## Alternatives
+
+### 1. ServiceBinding for attaching Policies and Routes for Mesh
+A new ServiceBinding resource has been proposed for mesh use cases. This would
+provide a way to attach policies, including Routes to a Service.
+
+Most notably, these provide a way to attach different policies to requests
+coming from namespaces or specific Gateways. In the example below, a
+ServiceBinding in the consumer namespace would be applied to the selected
+Gateway and affect all requests from that Gateway to the foo Service. Beyond
+policy attachment, this would also support attaching Routes as policies, in this
+case the attached HTTPRoute would split requests between the foo-a and foo-b
+Service instead of the foo Service.
+
+![Simple Service Binding Example](images/713-servicebinding-simple.png)
+
+This approach can be used to attach a default set of policies to all requests
+coming from a namespace. The example below shows a ServiceBinding defined in the
+producer namespace that would apply to all requests from within the same
+namespace or from other namespaces that did not have their own ServiceBindings
+defined.
+
+![Complex Service Binding Example](images/713-servicebinding-complex.png)
+
+#### Advantages
+* Works well for mesh and any use cases where requests don’t always transit
+  through Gateways and Routes.
+* Allows policies to apply to an entire namespace.
+* Provides very clear attachment of polices, routes, and more to a specific
+  Service.
+* Works well for ‘shrink-wrap application developers’ - the packaged app does
+  not need to know about hostnames or policies or have extensive templates.
+* Works well for ‘dynamic’ / programmatic creation of workloads ( Pods,etc - see
+  CertManager)
+* It is easy to understand what policy applies to a workload - by listing the
+  bindings in the namespace.
+
+#### Disadvantages
+* Unclear how this would work with an ingress model. If Gateways, Routes, and
+  Backends are all in different namespaces, and each of those namespaces has
+  different ServiceBindings applying different sets of policies, it’s difficult
+  to understand which policy would be applied.
+* Unclear if/how this would interact with existing the ingress focused policy
+  proposal described below. If both coexisted, would it be possible for a user
+  to understand which policies were being applied to their requests?
+* Route status could get confusing when Routes were referenced as a policy by
+  ServiceBinding
+* Introduces a new mesh specific resource.
+
+### 2. Attaching Policies for Ingress
+An earlier proposal for policy attachment in the Gateway API suggested adding
+policy references to each Resource. This works very naturally for Ingress use
+cases where all requests follow a path through Gateways, Routes, and Backends.
+Adding policy attachment at each level enables different roles to define
+defaults and allow overrides at different levels.
```

Comment:
Based on @howardjohn 's comment: https://github.com/kubernetes-sigs/gateway-api/pull/715/files#r668252011

Did we consider having a structure of this sort:
```yaml
spec:
  overridableConfiguration:
  enforcedConfiguration:
```

The `default` word in a precedence order seems confusing. The above proposal is verbose but it reduces the chances of misunderstanding.

Reply:
This is out of scope for this GEP (as noted already in the doc) so please ignore it.

Reply:
I've tried to clear up the difference between `default` and `override` in terms of precedence in my latest commit. Hopefully that helps here. I'm also open to different name than `default` here but would prefer somewhat shorter names if possible.